tests/test_trailedSources.py

```diff
@@ -0,0 +1,207 @@
+#
+
+import numpy as np
+import unittest
+import lsst.utils.tests
+import lsst.meas.extensions.trailedSources
+from lsst.meas.base.tests import AlgorithmTestCase
+from lsst.utils.tests import methodParameters
+
+# Trailed-source length, angle, and centroid.
+np.random.seed(432)
+nTrails = 50
+Ls = np.random.uniform(2, 20, nTrails)
+thetas = np.random.uniform(0, 2*np.pi, nTrails)
+xcs = np.random.uniform(0, 100, nTrails)
+ycs = np.random.uniform(0, 100, nTrails)
+
+
+class TrailedSource:
+    """Holds a set of true trail parameters.
+    """
+
+    def __init__(self, instFlux, length, angle, xc, yc):
+        self.instFlux = instFlux
+        self.length = length
+        self.angle = angle
+        self.center = lsst.geom.Point2D(xc, yc)
+        self.x0 = xc - length/2 * np.cos(angle)
+        self.y0 = yc - length/2 * np.sin(angle)
+        self.x1 = xc + length/2 * np.cos(angle)
+        self.y1 = yc + length/2 * np.sin(angle)
+
+
+# "Extend" meas.base.tests.TestDataset
+class TrailedTestDataset(lsst.meas.base.tests.TestDataset):
+    """A dataset for testing trailed source measurements.
+    Given a `TrailedSource`, construct a record of the true values and an
+    Exposure.
+    """
+
+    def __init__(self, bbox, threshold=10.0, exposure=None, **kwds):
+        super().__init__(bbox, threshold, exposure, **kwds)
+
+    def addTrailedSource(self, trail):
+        """Add a trailed source to the simulation.
+        'Re-implemented' version of
+        `lsst.meas.base.tests.TestDataset.addSource`. Numerically integrates a
+        Gaussian PSF over a line to obtain am image of a trailed source.
+        """
+
+        record = self.catalog.addNew()
+        record.set(self.keys["centroid"], trail.center)
+        covariance = np.random.normal(0, 0.1, 4).reshape(2, 2)
+        covariance[0, 1] = covariance[1, 0]
+        record.set(self.keys["centroid_sigma"], covariance.astype(np.float32))
+        record.set(self.keys["shape"], self.psfShape)
+        record.set(self.keys["isStar"], False)
+
+        # Sum the psf at each
+        numIter = int(5*trail.length)
+        xp = np.linspace(trail.x0, trail.x1, num=numIter)
+        yp = np.linspace(trail.y0, trail.y1, num=numIter)
+        for (x, y) in zip(xp, yp):
+            pt = lsst.geom.Point2D(x, y)
+            im = self.drawGaussian(self.exposure.getBBox(), trail.instFlux,
+                                   lsst.afw.geom.Ellipse(self.psfShape, pt))
+            self.exposure.getMaskedImage().getImage().getArray()[:, :] += im.getArray()
+
+        record.set(self.keys["instFlux"], self.exposure.getImage().array.sum())
+        self._installFootprint(record, self.exposure.getImage())
+
+        return record, self.exposure.getImage()
+
+
+# Following from meas_base/test_NaiveCentroid.py
+# Taken from NaiveCentroidTestCase
+class TrailedSourcesTestCase(AlgorithmTestCase, lsst.utils.tests.TestCase):
+
+    @methodParameters(L=Ls, theta=thetas, xc=xcs, yc=ycs)
```

Comment:
I don't think that this was the intended behavior: 
 - The decorated `setUp` is called right before `testNaivePlugin` (and `testVeresPlugin`).
 - `methodParameters` loop through the inputs, calling the original `setUp` for each input.
 - `testNaivePlugin` (and `testVeresPlugin`) actually _runs only once_, with the result of the very last `setUp` call.

Reply:
This has been changed to use a `classParameters` decorator on the class itself. Thanks!